### PR TITLE
Improved error handling when fetching bucket prefix that contains more than 1000 objects. #180

### PR DESCRIPTION
**Pull request by eddiesarevalo**
## 🗒️ Summary

This merges the latest develop into main. https://github.com/NASA-PDS/s3-browser-cloudfront/pull/180

This PR addresses improvements to the codebase to resolve an infinite loop issue when attempting to paginate through the contents of bucket's prefix to get the full list of objects. The issue presents itself when the `marker` query param isn't provided to the bucket via the CF behavior.

1) When the code detects that it is not advancing to the next paginated list of contents at a prefix, the code will stop paginating and display what it has received up to that point.
2) A warning message is displayed when the codebase detects that pagination isn't advancing that the user should contact the PDS Operator to report the issue.

image

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [x] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [ ] AI generated substantial portions of this code

Estimated % of code influenced by AI: ___ %

## ⚙️ Test Data and/or Report

Tested locally. The following screenshot shows that the codebase no longer loops infinitely for prefixes that have more than 1000 objects and displays what it has fetched up to that point and displays the warning message at the bottom of the list.

image

## ♻️ Related Issues

Resolves #178 

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [x] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [x] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [x] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [x] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [x] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [x] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: src/js/list.js [hunk 1]
```diff
@@ -195,12 +196,27 @@ function getS3Data(marker, table) {
                 tbody.append(rows);
             }
             if (info.nextMarker) {
-                getS3Data(info.nextMarker, table);
+                if (marker && info.nextMarker === marker) {
```
Comment (nutjob4life) [inline src/js/list.js:199]: Same comment as for PR #180: 

Tiny maintenance issue: the `else` branch does the same thing as the true branch of the condition, namely:
```javascript
$('#files').append(table);
sortables_init();
render(SUBDIRS);
```
Maybe a little refactoring is called for in case a change is needed in one place and forgotten in the other?

**Review by nutjob4life — APPROVED** [review]
Delta: ✓
Sense of déjà vu: ✓
Copy/pasting a review comment from #180: ✓
And even after all that: ✅
(But see the copied comment which applies to this PR too.)

**Review by anilnatha — APPROVED** [review]
